install/kubernetes/cilium/templates/cilium-configmap.yaml

```diff
@@ -485,23 +485,21 @@ data:
   #   - vxlan (default)
   #   - geneve
 {{- if .Values.gke.enabled }}
-  {{- if ne (.Values.routingMode | default "native") "native" }}
+  {{- if ne .Values.routingMode "native" }}
     {{- fail (printf "RoutingMode must be set to native when gke.enabled=true" )}}
   {{- end }}
-  routing-mode: "native"
```

Comment:
So this is a breaking change in behavior. Previously users were able to install GKE with just `set gke.enabled=true`, but now they explicitly also have to set `--set routingMode=native`. There is already [docs](https://docs.cilium.io/en/stable/installation/k8s-install-helm/) and [tooling](https://github.com/cilium/cilium/blob/fbd276676e7659b986e468d1e428065ff2f9af2a/cilium-cli/install/helm.go#L75-L77) out there that doesn't do that, all of that will be broken and would have to be fixed.

Is there no other way were we can preserve the current behavior without breaking this?

Reply:
I could revert this line and add an `and` statement when checking for the routing-mode being set. We are lucky here that gke.enabled is the only exception that requires a hard-coded routing-mode.

Reply:
Sounds good to me!

Reply:
What do you think of my new proposal?

Reply:
enabling `aksbyocni` just enforces the default values. And you cannot overwrite the routing-mode with gke enabled. If you set it to `native` explicitly, it will just be set on line 491 (that's why I have to disable it on line 502, to not let it be set twice), other values are not allowed at all.